torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -78,24 +89,27 @@ bool isSimpleMap(Node *node) {
   TensorType* expected_type = node->inputs()[0]->type()->cast<TensorType>();
   if (!expected_type)
     return false;
-  static const auto equal_modulo_strides = [](TensorType* expected, const TypePtr& _actual) {
+  static const auto equal_modulo_strides = [](TensorType* expected, const TypePtr& _actual, bool checkScalarType) {
```

Comment:
Actually can you please remove this altogether? There's no point in checking IO types in `isSimpleMap`, since in the only place it's used it is `&&`ed with `allFloatIO`, which implies they match. It's simpler to just always do the `isSimpleMap` check, and later have special cases for types of different ops.